### PR TITLE
Allow setting defaults for histogram and summary options

### LINKED ISSUE #336: Missing support for default summary and histogram options
I'm able to configure default values for quantiles in the configuration file under `defaults`. However, `max_summary_age` and `summary_age_buckets` seem to be only defined within `mappings` for a specific metric. Is there a way to define these 2 parameters globally, for all metrics?

### PR DESCRIPTION
**Pull request by glightfoot**
The first PR got too complex as I found other little edge cases. This should be a simpler version.

This allows setting histogram_options and summary_options in the defaults section. I added some tests and refactored the mapper tests to use named subtests, but I still want to test it more thoroughly before I mark this as ready.

Feedback appreciated.

Fixes #336 @matthiasr

### CODE REVIEW

File: pkg/registry/registry.go [hunk 1]
```diff
@@ -295,14 +296,21 @@ func (r *Registry) GetSummary(metricName string, labels prometheus.Labels, help
...
 	if vh == nil {
 		metricsCount.WithLabelValues("summary").Inc()
-		quantiles := r.Mapper.Defaults.Quantiles
+		quantiles := r.Mapper.Defaults.SummaryOptions.Quantiles
 		if mapping != nil && mapping.SummaryOptions != nil && len(mapping.SummaryOptions.Quantiles) > 0 {
 			quantiles = mapping.SummaryOptions.Quantiles
 		}
-		summaryOptions := mapper.SummaryOptions{}
+
+		summaryOptions := mapper.SummaryOptions{
```
Comment (glightfoot) [inline pkg/registry/registry.go:304]: I'm not sure if it's strictly necessary to uses the defaults here, but this feels more correct since some people might use the registry without the mapper (I do this in a few projects). Users of the mapper or registry package will need to update their usage

**Review by matthiasr — APPROVED** [review]
This looks good! Let me know when you think it's ready to merge.

**Comment by glightfoot** [comment]
I still haven't had time to actually integration test this, but I think it works as intended. I'll mark it ready for review in case someone else wants to test it, as the next few weeks are going to be busy for me.

**Comment by matthiasr** [comment]
I gave it a quick run but I think the tests are pretty comprehensive for this 😄  thank you for implementing it!
